app/assets/javascripts/hubstats/application.js

```diff
@@ -14,45 +14,85 @@
 //= require jquery_ujs
 //= require select2
 //= require hubstats/bootstrap
+//= require bootstrap-datepicker
 //= require_tree .
 
 $(document).ready( function() {
+  $('.input-daterange').datepicker({
+    "todayHighlight": true,
+    "endDate": "Today",
+    "todayBtn": true
+  });
   setDateRange()
 });
 
 function setDateRange() {
-  var index = readCookie("hubstats_index") || 2;
-  var timer = document.getElementById("time-select");
+  var dates;
+  var timer;
+  var datesArray = [];
 
-  timer.selectedIndex = index;
+  if ((readCookie("hubstats_dates") === "null~~null") || (readCookie("hubstats_dates") === null)) {
+    dates = getDefaultDateRange();
+  } else {
+    dates = readCookie("hubstats_dates")
+  }
 
-  timer.onchange = function() {
-    createCookie("hubstats_index",this.selectedIndex,1);
+  timer = document.getElementById("submitDateRange");
+  datesArray = dates.split("~~")
+
+  $('.input-daterange').find('[name="start"]').datepicker('update', new Date(datesArray[0]));
+  $('.input-daterange').find('[name="end"]').datepicker('update', new Date(datesArray[1]));
+
+  timer.onclick = function() {
+    var start_date;
+    var end_date;
+    var date;
+    start_date = $('.input-daterange').find('[name="start"]').datepicker('getDate');
+    end_date = $('.input-daterange').find('[name="end"]').datepicker('getDate');
+    
+    date = new Date();
+    date.setTime(date.getTime()+(24*60*60*1000));
+    var expires = "; expires="+date.toGMTString();
+    createCookie("hubstats_dates=" + start_date + "~~" + end_date + expires + "; path=/");
     window.location.reload();
   };
 };
 
-function createCookie(name,value,days) {
-    if (days) {
-        var date = new Date();
-        date.setTime(date.getTime()+(days*24*60*60*1000));
-        var expires = "; expires="+date.toGMTString();
-    }
-    else var expires = "";
-    document.cookie = name+"="+value+expires+"; path=/";
+function createCookie(str) {
+  document.cookie = str;
```

Comment:
You still need the path and expiration stuff around the value to be a valid cookie.  Same with readCookie.  LIke ti was before.